src/qvm.lisp

```diff
@@ -4,101 +4,144 @@
-  "Make a new quantum virtual machine with NUM-QUBITS number of qubits and a classical memory size of CLASSICAL-MEMORY-SIZE bits.
-
-ALLOCATION is an optional argument with the following behavior.
-
+  "Make a new quantum virtual machine with NUM-QUBITS number of qubits and a classical memory size of CLASSICAL-MEMORY-SIZE bits.ALLOCATION is an optional argument with the following behavior.
     - If it's NULL (default), then a standard wavefunction in the Lisp heap will be allocated.
-
     - If it's a STRING, then the wavefunction will be allocated as a shared memory object, accessible by that name.
-
-    - Otherwise, it's assumed to be an object that is compatible with the ALLOCATION-LENGTH and ALLOCATE-VECTOR methods
-"
+    - Otherwise, it's assumed to be an object that is compatible with the ALLOCATION-LENGTH and ALLOCATE-VECTOR methods"
   (check-type num-qubits unsigned-byte)
   (check-type classical-memory-model quil:memory-model)
-  (let ((allocation
-          (etypecase allocation
-            (null
-             (make-instance 'lisp-allocation :length (expt 2 num-qubits)))
-            (string
-             (make-instance 'posix-shared-memory-allocation :length (expt 2 num-qubits)
-                                                            :name allocation))
-            (t
-             (assert (= (allocation-length allocation) (expt 2 num-qubits)))
-             allocation))))
-    (multiple-value-bind (amplitudes finalizer)
-        (allocate-vector allocation)
-      ;; Ensure we start in |...000>.
-      (bring-to-zero-state amplitudes)
-      ;; Make the QVM.
-      (let ((qvm (make-instance 'pure-state-qvm
-                                :number-of-qubits num-qubits
-                                :amplitudes amplitudes
-                                :classical-memory-subsystem
-                                (make-instance 'classical-memory-subsystem
-                                               :classical-memory-model
-                                               classical-memory-model))))
-        ;; When the QVM disappears, make sure the shared memory gets
-        ;; deallocated too. XXX: This could cause problems if someone
-        ;; copies out the shared memory. We should enforce the use of
-        ;; MAP-AMPLITUDES and AMPLITUDE-REF and the like.
-        (tg:finalize qvm finalizer)
-        ;; Return the QVM.
-        qvm))))
+  (make-instance 'pure-state-qvm
+                 :number-of-qubits num-qubits
+                 :state (make-pure-state num-qubits :allocation allocation)
+                 :classical-memory-subsystem
+                 (make-instance 'classical-memory-subsystem
+                                :classical-memory-model
+                                classical-memory-model)))
 
 (defmethod compile-loaded-program ((qvm pure-state-qvm))
-  "Compile the loaded program on the QVM QVM."
+  ;; Compile the loaded program on the PURE-STATE-QVM QVM.
```

Comment:
Why not keep this comment as a docstring?

Reply:
I thought `defmethods` were supposed to be commented (instead of docstrings)